providers/src/airflow/providers/jdbc/hooks/jdbc.py

```diff
@@ -25,6 +25,7 @@
 import jaydebeapi
 import jpype
 from sqlalchemy.engine import URL
+from wrapt import synchronized
```

Comment:
I have not found "synchronized" as something that wrapt would publish as an interface. It seems it is somewhat accidentally available in wrapt implementation and might change any time. 

I think we should find a more "official" tool to doing it.

Reply:
Hmm. I found it in https://wrapt.readthedocs.io/en/master/examples.html - but search did not found it. It's therefore an example in wrapt - which is also a bit strange to rely on ?

Reply:
Possibly better to use with 

```
self.lock = threading.RLock()

with self.lock:
     jaydebeapi.connect( ...
```

It's essentially the same but better expresses the fact that we want to synchronize jaydebeapi, not the `get_conn` method.

Reply:
> Hmm. I found it in https://wrapt.readthedocs.io/en/master/examples.html - but search did not found it. It's therefore an example in wrapt - which is also a bit strange to rely on ?

I liked the solution as to me, coming from Java, it seemed like a nice a clean solution.  You can use the decorator as an aspect which prevents other threads from entering until the current one has finished, no need to hard code this into the function as this is a crosscutting concern, imho.

Reply:
> Possibly better to use with
> 
> ```
> self.lock = threading.RLock()
> 
> with self.lock:
>      jaydebeapi.connect( ...
> ```
> 
> It's essentially the same but better expresses the fact that we want to synchronize jaydebeapi, not the `get_conn` method.

I will adapt it like this then.

Reply:
> I liked the solution as to me, coming from Java, it seemed like a nice and clean solution.

Yeah. It's nice, but it's more of a byproduct of what they have as an example usage. What happens if one day they release the package and the "synchronized" decorator will not be there or moved? 

They can do it even in patchlevel version. and claim "it's ok to move it - it's just an example" - we cannot rely on it being there. 

We could also write our own decorator (basically copy what they have done) and keep it in, but that seems an overkill if we can do the same with `self.lock =" and `with lock` two lines added. 

It's abit like infamous `leftpad` fiasco.